pkg/cmd/roachtest/tests/mixed_version_cdc.go

```diff
@@ -343,12 +345,47 @@ func (cmvt *cdcMixedVersionTester) createChangeFeed(node int) versionStep {
 			{"updated", ""},
 			{"resolved", fmt.Sprintf("'%s'", resolvedInterval)},
 		}
-		_, err := newChangefeedCreator(db, fmt.Sprintf("%s.%s", targetDB, targetTable), cmvt.kafka.sinkURL(ctx)).
-			With(options...).
-			Create()
+		creator := newChangefeedCreator(db, fmt.Sprintf("%s.%s", targetDB, targetTable), cmvt.kafka.sinkURL(ctx)).
+			With(options...)
+
+		jobID, err := creator.Create()
 		if err != nil {
 			t.Fatal(err)
 		}
+		// In 22.1, a changefeed may fail during a rolling upgrade due to a bug
+		// which is fixed in 22.2 onwards. To accomodate for this bug,
+		// poll the changefeed job and restart it if it fails.
+		// See #106878 for more details.
+		go watchChangefeed(ctx, t, jobID, db, creator)
```

Comment:
Let's replace this with `cmvt.monitor.Go()` instead. `go` is discouraged in roachtests as a panic in the goroutine would lead to the entire roachtest process crashing (unless you add a `recover` to all of them, which is easy to forget).